### PR TITLE
created main chat page

### PR DESCRIPTION
**Pull request by lestraza**

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
-    return <div className="wrapper">
-        <Main/>    
-    </div>;
+    return (
+      <div className="wrapper">
+        <div className="header"></div>
+        <div className="main"></div>
+        <div className="app_container">
+          <ChatList />
+          <Dialog name={"wefew"} />
```
Comment (stahlhammer88) [inline client/src/App.tsx:14]: remove the `name` prop

File: client/src/components/ChatPreview.tsx [hunk 2]
```diff
@@ -9,7 +9,9 @@ export interface IChatPreviewProps {
 export default class ChatPreview extends React.Component<IChatPreviewProps> {
   public render() {
     return (
-      <div>
+      <div className='chat_preview'>
+        <div className='user_pic'>{this.props.user.avatarUrl}</div>
```
Comment (stahlhammer88) [inline client/src/components/ChatPreview.tsx:13]: Use this as an actual url either in `` or in `background-image`

File: client/src/components/ChatList.tsx [hunk 3]
```diff
@@ -18,11 +18,14 @@ export default class ChatList extends React.Component<IChatListProps> {
   public renderChatPreviews() {
       const { users } = this.mainStore
       return users.map((user, i) => {
-          <ChatPreview user={user}/>
+        return  <ChatPreview user={user}/>
```
Comment (stahlhammer88) [inline client/src/components/ChatList.tsx:21]: `=> <ChatPreview user={user}/>`

File: client/src/components/Dialog.tsx [hunk 4]
```diff
@@ -1,16 +1,50 @@
...
+  @action.bound
+  onChangeSaveValue(e: React.FormEvent<HTMLInputElement>) {
+    const { saveNewMessage } = this.mainStore
+    const value = e.currentTarget.value
+    saveNewMessage(value)
+  }
   public render() {
-      this.props.name
+      const { newMessage } = this.mainStore
+      console.log(newMessage);
```
Comment (stahlhammer88) [inline client/src/components/Dialog.tsx:33]: remove log

File: client/src/components/Message.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { inject, observer } from "mobx-react";
+import MainStore, { IMessage } from "../store/MainStore";
+import { computed } from "mobx";
+
+export interface IMessageProps {
+    message: IMessage
+}
+
+@inject("store")
```
Comment (stahlhammer88) [inline client/src/components/Message.tsx:10]: better use single quotes

File: client/src/components/Message.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  @computed
+  private get currentDate() {
+    const { message } = this.props;
+    const date = new Date(message.timeStamp)
+    const formatedDate = date.toLocaleString('ru', {
+        hour: '2-digit',
+        minute: '2-digit'
+    })
+    return formatedDate
```
Comment (stahlhammer88) [inline client/src/components/Message.tsx:34]: you don't need `formatedDate`, just `return date.toLocaleString...`

File: client/src/components/Message.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  private get isOutgoing() {
+      const { clientId } = this.mainStore
+      const { message } = this.props;
+      return message.userId === clientId
+  }
+
+  @computed
+  private get currentDate() {
+    const { message } = this.props;
+    const date = new Date(message.timeStamp)
```
Comment (stahlhammer88) [inline client/src/components/Message.tsx:29]: Weird destructuring, it's better either
```suggestion
    const { timeStamp} = this.props.message;
    const date = new Date(timeStamp)
```
or
```suggestion
    const date = new Date(this.props.message.timeStamp)
```

File: client/src/components/Message.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    const { message } = this.props;
+    return (
+        <div className={`message_screen_container ${this.isOutgoing? 'outgoing' : 'incoming'}`}>
+          <div className="message_screen_container__user_name">
+            {message.fullName}
+          </div>
+          <div className="message_screen_container__message">
+            {message.text}
+            <div className="message_screen_container__time_stamp">
+              {this.currentDate}
```
Comment (stahlhammer88) [inline client/src/components/Message.tsx:47]: you can destructure all these from `message`

File: client/src/components/MessageScreenContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
+import MainStore from "../store/MainStore";
+import { action } from "mobx";
+import { inject, observer } from "mobx-react";
+import Message from "./Message";
+
+export interface IMessageScreenContainerProps {}
+
+@inject("store")
```
Comment (stahlhammer88) [inline client/src/components/MessageScreenContainer.tsx:9]: quotes

File: client/src/components/Dialog.tsx [hunk 10]
```diff
@@ -1,16 +1,50 @@
-import React, { Component} from 'react';
+import React, { Component, FormEventHandler} from 'react';
+import { action } from 'mobx';
+import { inject, observer } from 'mobx-react';
+import MainStore from '../store/MainStore';
+import MessageScreenContainer from './MessageScreenContainer';
 
 export interface IDialogProps {
     name: string
```
Comment (stahlhammer88) [inline client/src/components/Dialog.tsx:8]: you don't need this `name` here

File: client/src/store/MainStore.ts [hunk 11]
```diff
@@ -1,33 +1,65 @@
...
+      this.getUsers()
+  }
+
+  @action.bound
+  public getUsers() {
+    const newUser = {
+      id: "35s4d354",
+      name: "Erjan",
+      lastName: "Egoev",
+      avatarUrl: "sldkjfls",
```
Comment (stahlhammer88) [inline client/src/store/MainStore.ts:42]: you can copy an actual image orl from google

File: client/src/App.tsx (hunk 1 above)
Comment (lestraza) [reply client/src/App.tsx:14]: done

File: client/src/components/ChatPreview.tsx (hunk 2 above)
Comment (lestraza) [reply client/src/components/ChatPreview.tsx:13]: done

File: client/src/components/ChatList.tsx (hunk 3 above)
Comment (lestraza) [reply client/src/components/ChatList.tsx:21]: done

File: client/src/components/Dialog.tsx (hunk 4 above)
Comment (lestraza) [reply client/src/components/Dialog.tsx:33]: done

File: client/src/components/Message.tsx (hunk 5 above)
Comment (lestraza) [reply client/src/components/Message.tsx:10]: ok

File: client/src/components/Message.tsx (hunk 6 above)
Comment (lestraza) [reply client/src/components/Message.tsx:34]: ok

File: client/src/components/Message.tsx (hunk 7 above)
Comment (lestraza) [reply client/src/components/Message.tsx:29]: ok, have done by the first way

File: client/src/components/Message.tsx (hunk 8 above)
Comment (lestraza) [reply client/src/components/Message.tsx:47]: ok

File: client/src/components/MessageScreenContainer.tsx (hunk 9 above)
Comment (lestraza) [reply client/src/components/MessageScreenContainer.tsx:9]: ok

File: client/src/components/Dialog.tsx (hunk 10 above)
Comment (lestraza) [reply client/src/components/Dialog.tsx:8]: ok

File: client/src/store/MainStore.ts (hunk 11 above)
Comment (lestraza) [reply client/src/store/MainStore.ts:42]: ok

**Pull request merged**